### DISCUSSION TITLE
nnUNet with MONAI for png images problem 😭

### DISCUSSION THREAD

**Discussion by swhoosh** [Project-MONAI/tutorials#1594] [Q&A]
Hi guys. I'm trying to use nnUNet on MONAI since I think it's easier to deploy later with MONAI deploy. But, I am currently having a problem converting my dataset. My dataset is in `.png` format and I am trying to convert it into nnunet format using this command line:
```
python -m monai.apps.nnunet nnUNetV2Runner convert_dataset --input_config="./input.yaml"
```
My `input.yaml` looks like this:
```
modality: CT
datalist: "/ai_neuro/swoosh/dataset/Dataset413_muscle/Dataset413_muscle_folds.json"
dataroot: "/ai_neuro/swoosh/dataset/Dataset413_muscle"
dataset_name_or_id: 413
```
But somehow the conversion returns error saying:
```
python -m monai.apps.nnunet nnUNetV2Runner convert_dataset --input_config="./input.yaml"
2023-12-15 16:34:42,330 - WARNING - Dataset with name/ID: 413 cannot be found in the record. Please ignore the message above if you are running the pipeline from a fresh start. But if the dataset is expected to be found, please check your input_config.
2023-12-15 16:34:42,352 - INFO - num_input_channels: 1
2023-12-15 16:34:43,210 - INFO - num_foreground_classes: 1
2023-12-15 16:34:43,210 - INFO - converting data section: training...
  0%|                                                                                                                               | 0/371 [00:00<?, ?it/s]
2023-12-15 16:34:43,222 - WARNING - Input config may be incorrect. Detail info: error/exception message is:
 'original_affine'
```
Which seems weird so I try investigating more on what were generated. Here's what I found.
1. `work_dir` which is working directory for nnUNet is already generated.
2. `dataset.json` had incorrect file extension (should have been .png).
```
{
    "channel_names": {
        "0": "CT"
    },
    "file_ending": ".nii.gz",
    "labels": {
        "background": 0,
        "class1": 1
    },
    "numTraining": 371
}
```
3. Other folders have been created but empty inside.
4. I have tried looking through docs already and it seems that `analyze_data` might be the problem since it only detects 1 input channel. However, I'm unable to find the source code of this function `from monai.apps.nnunet.utils import analyze_data, create_new_data_copy, create_new_dataset_json`
5. I have tried using the native nnUNetv2 and it works fine. I am able to use `.png` with RGB channels to train my model successfully. 

This is my how my dataset folder looks like:
```
-rwx------ 1 root root 93660 Dec 15 13:35 Dataset413_Sarcopenia_folds.json
-rw-r--r-- 1 root root   245 Dec 15 13:34 dataset.json
drwxr-xr-x 1 root root     0 Dec 15 13:34 imagesTr
drwxr-xr-x 1 root root     0 Dec 15 13:34 imagesTs
drwxr-xr-x 1 root root     0 Dec 15 13:34 labelsTr
drwxr-xr-x 1 root root     0 Dec 15 13:34 labelsTs

#  in Dataset413_Sarcopenia_folds.json
{
  "testing": [...]
  "training": [
          {
              "image": "/ai_neuro/swoosh/dataset/Dataset413_muscle/imagesTr/23452834.png",
              "label": "/ai_neuro/swoosh/dataset/Dataset413_muscle/labelsTr/23452834.png",
              "fold": 0
          },
          {
              "image": "/ai_neuro/swoosh/dataset/Dataset413_muscle/imagesTr/24112438.png",
              "label": "/ai_neuro/swoosh/dataset/Dataset413_muscle/labelsTr/24112438.png",
              "fold": 0
          },
          ...
  ]
}
```

Anyone has any ideas for this? Thank you so much in advance.

Here's the reference I have been digging through. https://docs.monai.io/en/stable/_modules/monai/apps/nnunet/nnunetv2_runner.html#nnUNetV2Runner.convert_dataset

**Comment by KumoLiu** [comment]
Hi @swhoosh, thanks for reporting.
The reason is that we hard-coded here. 
https://github.com/Project-MONAI/MONAI/blob/90d2acb54d5325e6727f2cc2329cb40adc8577af/monai/apps/nnunet/utils.py#L166
https://github.com/Project-MONAI/MONAI/blob/90d2acb54d5325e6727f2cc2329cb40adc8577af/monai/apps/nnunet/utils.py#L104

I will try to fix it.
Thanks.
cc @dongyang0122 @mingxin-zheng

**Reply by swhoosh** [reply]
Thank you so much for your hard work!
